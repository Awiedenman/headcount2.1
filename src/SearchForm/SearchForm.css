.search {
  display: flex;
  justify-content: center;
}

input {
  width: 300px;
  padding: 2%;
  font-size: 1em;
}