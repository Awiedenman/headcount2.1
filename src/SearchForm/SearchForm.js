import React, { Component } from 'react';

export class SearchForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      searchInput: ''
    }
  }

  handleSearch = (districtName) => {
    // this.props.searchDistricts(districtName)
  }

  handleChange = (e) => {
    this.setState({
      searchInput: e.target.value
    })
  }

  render() {
    return (
      <form onSubmit={() => {
        this.handleSearch(this.state.searchInput)
      }} className="search">
        <input 
          type="text" 
          name="search" 
          placeholder="search for area"
          onChange={this.handleChange}
        />
      </form>
    )
  }
}