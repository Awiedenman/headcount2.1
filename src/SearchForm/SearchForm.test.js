import React from 'react';
import { SearchForm } from './SearchForm';
import { shallow } from 'enzyme';

describe('SearchForm', () => {
  it('matches the snapshot of the SearchForm', () => {
    const mockShowSearchResults = jest.fn();
    const wrapper = shallow( 
      <SearchForm 
        showSearchResults={ mockShowSearchResults }
      />);

    expect(wrapper).toMatchSnapshot();
  });
});