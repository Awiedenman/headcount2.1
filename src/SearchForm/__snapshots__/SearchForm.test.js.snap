// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm matches the snapshot of the SearchForm 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    showSearchResults={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        name="search"
        onChange={[Function]}
        placeholder="search for area"
        type="text"
/>,
      "className": "search",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "name": "search",
        "onChange": [Function],
        "placeholder": "search for area",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          name="search"
          onChange={[Function]}
          placeholder="search for area"
          type="text"
/>,
        "className": "search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "search",
          "onChange": [Function],
          "placeholder": "search for area",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
